File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> s;
        for (int num: nums) {
            s.insert(num);
        }        
        set<int> ps;
        for (int num: nums) {
            if (s.find(num - 1) == s.end()) {
                ps.insert(num);
            }
        }
        int res = 0;
        for (int num: ps) {
            int curr = num;
            int len = 1;
            while (s.find(curr + 1) != s.end()) {
                ++curr;
                ++len;
            }
            res = max(res, len);
        }
        return res;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def anagramDict(self, s: str) -> dict[str, int]:
        res = {}
        for c in s:
            res[c] = res.get(c, 0) + 1
        return res
    def isAnagram(self, s: str, t: str) -> bool:
        return self.anagramDict(s) == self.anagramDict(t)